cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfOperationsCreator.java

```diff
@@ -64,12 +67,20 @@ private static ReactorCloudFoundryClient createCloudFoundryClient(
     private static PasswordGrantTokenProvider createTokenProvider(
             LoginCommandOptions commandOptions) {
 
+        String user = determineCredential(commandOptions.getUserName(), CF_CONTROL_USER);
+        String password = determineCredential(commandOptions.getPassword(), CF_CONTROL_PASSWORD);
+
         return PasswordGrantTokenProvider.builder()
-                .password(commandOptions.getPassword())
-                .username(commandOptions.getUserName())
+                .username(user)
+                .password(password)
                 .build();
     }
 
+    private static String determineCredential(String credential, String environmentVariableName) {
+        String environmentVariableValue = System.getenv(environmentVariableName);
```

Comment:
The call of getenv might throw a SecurityException (see [here](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#getenv-java.lang.String-)). If this case can ever occur, then there should be an according "@throws" entry in the documentation of the constructor, or the exception should be caught in here and handled properly.